### PR TITLE
Feature/new service

### PR DESCRIPTION
**Pull request by aaron-launch**
**_launch-cli_** commands to to create a new service

This command creates a new opinionated service utilizing a skeleton repo. When this command is run, it clones a specified or default skeleton repository, then creates a new repository from this skeleton. Once created, it creates all the folders within this repository from the supplied inputs. Inputs must match the skeleton repo `file_structure.json`. **_There is currently no validation yet to confirm the supplied inputs match the expected structure from the skeleton_**. Once the folders are created, the command will loop through the jinja templates and place them in the desinated spot listed in the `file_structure.json`. After all this is completed, It pushes are changes to remote. 

Default skeleton: 

### Example
```launch service create --name my-demo-svc```

### Command Accepted Arguments 
`--organization`GitHub organization containing your repository. Defaults to the nexient-llc organization.
`--name` (Required) Name of the service to  be created.
`--description` A short description of the repository. Default: 'A short description of the repository.'
`--skeleton-url` A skeleton repository url that this command will utilize during this creation.
`--skeleton-branch` The branch or tag to use from the skeleton repository.
`--public`The visibility of the repository.
`--visibility` The visibility of the repository. Can be one of: public, private.
`--main-branch` Working directory path. Defaults to home (~).
`--init-branch` The name of the main branch.
`--in-file`
`--dry-run`Perform a dry run that reports on what it would do, but does not create webhooks.

### CODE REVIEW

**Review by dsahoo-nexient — APPROVED** [review]

File: src/launch/service/common.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    print(f"{template_name} : {type(template_name)}")
+    print(output_path)
+    print(data)
+    template = env.get_template(template_name)
+    rendered_content = template.render(data={
+        "path": output_path,
+        "config": data
+    })
+
+    # Regular expression to extract output file name
```
Comment (chris-taylor-ntt) [inline src/launch/service/common.py:34]: Left a comment on the skeleton PR with some details on how you might want to restructure this.

File: src/launch/service/common.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import logging
+import os
```
Comment (chris-taylor-ntt) [inline src/launch/service/common.py:2]: Consider swapping usage of os.path* in this file for pathlib.

File: src/launch/cli/service/commands.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    help="A short description of the repository.",
+)
+@click.option(
+    "--skeleton-url",
+    default=SERVICE_SKELETON,
+    help="A skeleton repository url that this command will utilize during this creation.",
+)
+@click.option(
+    "--skeleton-branch",
+    default=SERVICE_SKELETON,
```
[the remark this answers is not in the document]
Comment (aaron-launch) [reply src/launch/cli/service/commands.py:42]: This was a bug actually. Should have been `SKELETON_BRANCH`

File: src/launch/service/common.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                raise Exception(f"File not found: {value}")
+        elif isinstance(value, dict):
+            new_path = os.path.join(base_path, key)
+            if not os.path.exists(new_path):
+                os.makedirs(new_path)
+            create_dirs_and_copy_files(new_path, value)
+
+
+def render_template(template_name, output_path, data, env):
+    print(f"{template_name} : {type(template_name)}")
```
[the remark this answers is not in the document]
Comment (aaron-launch) [reply src/launch/service/common.py:25]: These were old debug statements. removed.

**Pull request closed without merging**